backend/apps/events/models.py

```diff
@@ -0,0 +1,205 @@
+
+
+class Event(models.Model):
+    """
+    This class represents an Event which can be held by Recruiters (or generally
+    by staff members), where they can invite other staff members and candidates.
+
+    Attributes
+    ----------
+    title : str
+        a title of the event
+    description : str
+        a description of the event
+    event_type : EventType
+        a type of the event, which is defined by EventType model
+    priority : int
+        a priority of the event, defined by enum PriorityChoices
+    owner : Profile
+        an owner of the event (aka contact person)
+    staff_participants : Profile
+        staff members that will (or planning to) participate in the event
+    candidate_participants : Candidate
+        candidates that will (or planning to) participate in the event
+    start_time : datetime
+        start time of the event
+    end_time : datetime
+        end time of the event
+    duration : timedelta
+        duration of the event, it is being calculated on creation/update
+    created_at : datetime
+        time of creation of the event
+    changed_at : datetime
+        last time the event was changed
+    status : int
+        shows status of the event, enum EventStatusChoices
+    is_active : bool
+        returns True if event is active
+    is_completed : bool
+        returns True if event is COMPLETED
+    cancelled : bool
+        returns True if event is CANCELLED
+
+    Methods
+    ----------
+    mark_as_active():
+        marks event as ACTIVE
+    mark_as_completed():
+        marks event as COMPLETED
+    mark_as_cancelled():
+        marks event as CANCELLED
+
+    TODO
+    ----------
+    * After adding Celery, we can automatically put events to COMPLETED
+    if time is up...
+    * We can do an automatic email notification if the event was cancelled
+    (Celery + signals)
+
+    """
+
+    class Meta:
+        ordering = ("start_time",)
+        verbose_name = _("Event")
+        verbose_name_plural = _("Events")
+        get_latest_by = ("created_at",)
+
+        constraints = (
+            models.CheckConstraint(
+                check=models.Q(start_time__lt=models.F("end_time")),
+                name="start_before_end",
+                violation_error_message="Start time should be arlier than end time.",
+            ),
+        )
+
+    title = models.CharField(
+        _("Event Title"),
+        max_length=50,
+        db_index=True,
+    )
+    description = models.TextField(
+        _("Event Description"),
+        max_length=2048,
+    )
+    event_type = models.ForeignKey(
+        to=EventType,
+        related_name="events",
+        on_delete=models.SET_NULL,
+        blank=True,
+        null=True,
+    )
+    priority = models.IntegerField(
+        _("Priority level"),
+        choices=PriorityChoices.choices,
+        default=0,
+    )
+    owner = models.ForeignKey(
+        "accounts.Profile",
+        related_name="events_owner",
```

Comment:
The reverse relation is from Profile to Event, so maybe `related_name` should be 'owner_events' or 'events'?